CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -70,9 +72,7 @@ class CardClient internal constructor(
         try {
             cardAPI.fetchCachedOrRemoteClientID()
         } catch (e: PayPalSDKError) {
-            approveOrderListener?.onApproveOrderFailure(
-                APIClientError.clientIDNotFoundError(e.code, e.correlationID)
-            )
+            notifyApproveOrderFailure(APIClientError.clientIDNotFoundError(e.code, e.correlationID))
             return
```

Comment:
Let the error be handled by the `coroutineExcpetionHandler`:
```suggestion
            throw APIClientError.clientIDNotFoundError(e.code, e.correlationID)
```